### PR TITLE
REWORK CHANNEL (BACKEND):

### PR DESCRIPTION
**Pull request by riblanc**
Refactoring the join channel method to simplify the logic, performing fewer database accesses.
This new logic will subsequently allow us to be able to join a socket.io room specifying a socket.id, more easily.

### CODE REVIEW

**Comment by riblanc** [comment]
I'd like to review this PR with you @AdrianWeaver as I changed most of the join channel method.
I still apply the same logic as you did, but making fewer database calls, less if statements and removing redundant code.

See you at school!

File: back/src/chat/channel/channel.service.ts [hunk 1]
```diff
@@ -12,143 +8,152 @@ import {
...
+      ...pagination,
+    });
   }
 
-  //TODO: Fix this garbage function
-  isPasswordOk(channel: Channel, data: ChannelDto): boolean {
-    if (channel.password === data.password) return true;
-    return false;
+  async isUserInChannel(userId: number, channelId: number): Promise<boolean> {
+    return !!(await this.getChannelConnection(userId, channelId));
```
Comment (AdrianWeaver) [inline back/src/chat/channel/channel.service.ts:53]: Something like that maybe?
```suggestion
    const channelConnection = await this.getChannelConection(userId, channelId));
    if (!channelConnection || channelConnection.role === channelRole.INVITED
      || channelConnection.role ==== channelRole.BANNED)
      return false
    return true;
```

**Review by AdrianWeaver — CHANGES_REQUESTED** [review]

File: back/src/chat/channel/channel.service.ts (hunk 1 above)
Comment (AdrianWeaver) [reply back/src/chat/channel/channel.service.ts]: updated in : [4da690d](https://github.com/42rave/transcendence/pull/36/commits/4da690d473b70ada9865cb0e1fdf933ac74956ca)

**Comment by AdrianWeaver** [comment]
Let's all pretend you did not see me push --force on this branch :innocent:

**Pull request merged**